localflavor/bg/validators.py

```diff
@@ -0,0 +1,68 @@
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from .utils import get_egn_birth_date
+
+
+def egn_validator(egn):
+    '''
+    Check Bulgarian unique citizenship number (EGN) for validity
+    More details https://en.wikipedia.org/wiki/Unique_citizenship_number
+    Full information in Bulgarian about algoritm is available here
+    http://www.grao.bg/esgraon.html#section2
+
+    '''
+    def check_checksum(egn):
+        _weights = (2, 4, 8, 5, 10, 9, 7, 3, 6)
+        try:
+            checksum = sum(weight * int(digit) for weight, digit in zip(_weights, egn))
+            return int(egn[-1]) == checksum % 11 % 10
```

Comment:
Reason for doing double modulus operator are points 3. and 4. for calculating the checksum:

   1. Each digit is multiplied by its weight (see below)
   2. The products obtained are added
   3. The sum is divided by 11 ( use sum % 11, not /, modulus, not division)
   4. If the remainder is less than 10, the remainder is the checksum digit, otherwise the checksum digit is zero

When I first do modulus by 11 then result is not greater than 10 and instead of checking that I can just calculate modulus by 10.